stdlib/Serialization/src/Serialization.jl

```diff
@@ -1454,6 +1454,46 @@ function deserialize_expr(s::AbstractSerializer, len)
     resolve_ref_immediately(s, e)
     e.head = deserialize(s)::Symbol
     e.args = Any[ deserialize(s) for i = 1:len ]
+
+    # Rewrite old :method expressions to define_method calls
```

Comment:
Seems unlikely we should encounter or support this, since pre-lowered top level code is a pretty buggy concept even in a single session

Reply:
We have a specific test for this case, so I added the compat. We can remove the test if course